app/models/search/token.rb

```diff
@@ -106,36 +110,48 @@ def comparison(qual, op, rhs_or_values)
       end
     end
 
-    # generates an sql where clause fragment for a comparison with the given qualifier, operator, and value token
-    def comparison_fragment(qual, op, value_token)
+    def qualifier_has_more_than_one_column(qualifier)
```

Comment:
This should be a method on the Qualifier class. I have noticed this in your code several times. Think twice when writing a method on a class that does a small operation on a property of another class. Can it be made into a computed property of that second class?